tests/pybind11_test_mod.cc

```diff
@@ -41,6 +41,8 @@ bool acceptArray1(ndarray::Array<double,1,1> const & a1) {
 
 void acceptArray10(ndarray::Array<double,1,0> const & a1) {}
 
+void acceptArray11(ndarray::Array<double,1,1> const & a1) {}
```

Comment:
I have no one to blame but myself, but this all started to go wrong when that first `acceptArrayXY` snuck in with no meaningful implementation, as when I reviewed the last PR I certainly wasn't expecting this to be what was in the body of it.  And thinking about it more I better how this passes the test without really getting ndarray to the point of supporting zero shape elements.

I should probably try to find the time to make the behavior of those arrays just more consistent generally, but I really don't have time now.  In the meantime, could you maybe just rename `acceptArray10` and `acceptArray11` to something like `acceptZeroShapeArrayXY` to make it clear that they're more different from the other `acceptArray` methods?

Reply:
The naming of these is certainly confusing.  I thought that `accept` meant "pybind11 will allow this as an override" but some of the other ones are actually checking values.  But I'll rename right now